packages/react/src/firestore/useFirestoreDocument.test.tsx

```diff
@@ -51,4 +51,98 @@ describe("useFirestoreDocument", () => {
     expect(snapshot.exists()).toBe(true);
     expect(snapshot.data()?.foo).toBe("bar");
   });
+
+  test("fetches document from server source", async () => {
+    const ref = doc(firestore, "tests", "serverSource");
+
+    // set data
+    await setDoc(ref, { foo: "fromServer" });
+
+    //test the hook
+    const { result } = renderHook(
+      () =>
+        useFirestoreDocument(ref, {
+          queryKey: ["server", "doc"],
+          firestore: { source: "server" },
+        }),
+      { wrapper }
+    );
+
+    // await the query
+    await waitFor(() => {
+      expect(result.current.isSuccess).toBe(true);
+    });
+
+    // snapshot should exist, data should be fetched from the server and should contain the correct data
+    const snapshot = result.current.data;
+    expect(snapshot?.exists()).toBe(true);
+    expect(snapshot?.data()?.foo).toBe("fromServer");
+  });
+
+  test("handles fetch errors", async () => {
+    const ref = doc(firestore, "nonExistentCollection", "nonExistentDoc");
+
+    const { result } = renderHook(
+      () =>
+        useFirestoreDocument(ref, {
+          queryKey: ["error", "doc"],
+        }),
+      { wrapper }
+    );
+
+    await waitFor(() => {
+      expect(result.current.isError).toBe(true);
+    });
+
+    expect(result.current.error).toBeDefined();
```

Comment:
Could we make a test utility here which does something along the lines of `expectFirestoreError(result.current.error, 'firebase-code')`, and both checks `.code` is defined and matches what is expected?  We could hit a false positive here with it being an error unrelated to the none existent doc.

Reply:
sure yes, i do agree. let me setup a utility to run that check